src/tiled/mapdocumentactionhandler.cpp

```diff
@@ -655,13 +666,17 @@ void MapDocumentActionHandler::updateActions()
     if (currentLayer) {
         if (currentLayer->asTileLayer()) {
             mActionSelectNone->setEnabled(!selection.isEmpty());
+            mActionAutocrop->setEnabled(true);
         } else if (currentLayer->asObjectGroup()) {
             mActionSelectNone->setEnabled(selectedObjectsCount  > 0);
+            mActionAutocrop->setEnabled(false);
         } else {
             mActionSelectNone->setEnabled(false);
+            mActionAutocrop->setEnabled(false);
         }
     } else {
         mActionSelectNone->setEnabled(false);
+        mActionAutocrop->setEnabled(false);
```

Comment:
It will be more readable to put one line below together with `mActionCropToSelection` like this:
```c++
mActionAutocrop->setEnabled(currentLayer && currentLayer->isTileLayer());
```
The `mActionSelectNone` needed all this special code only because the condition on whether it should be enabled or not varies depending on the type of the selected layer, but let's not "abuse" all those cases for setting the state of `mActionAutocrop`.